### ISSUE TITLE
README: Describe differences between lutaml-model and Shale, add instructions for migration to lutaml-model

### ISSUE THREAD

**Issue by ronaldtse** [lutaml/lutaml-model#33]

**Comment by opoudjis** [comment]
I have assigned https://github.com/metanorma/metanorma/pull/403 to Hassan for investigation...

**Comment by ronaldtse** [comment]
> Shale uses:
> 
> ```
> map_element "bibdata", using: { from: :bibdata_from_xml,
>                                           to: :bibdata_to_xml }
> ```
> 
> Lutaml::Model uses
> 
> ```
>  map 'bibdata', to: :bibdata, with: { to: :bibdata_to_xml, from: :bibdata_from_xml }
> ```

There is a misunderstanding that is not reflected in the README. In an `xml` block, there are `map_element` `map_attribute` and `map_content` methods. In key-value format blocks like `json` and `yaml`, there is only one `map` method. 

If we are not guarding between the differences of both, we need to raise an error if `map` is called inside `xml` and if those other methods are called inside `json` for example.

There is some work to do in configuring the serialization adapters (e.g. NokogiriAdapter) because it is only the top-level gem/code that shall dictate the type of adapters used. We should not specify particular types of adapters inside gems because this is deployment specific.

In addition, we should specify the default serialization adapters.

Inside a gem, it is only in `spec/spec_helper.rb` that we need to specify the configuration for spec testing if we want to change away from defaults.

We also don't want people to need to manually load the `require '...'` file for serialization adapters.

I'm imagining a block like:
`spec/spec_helper.rb`:
```ruby
Lutaml::Model::Config.configure do |config|
  config.xml_adapter_type = :nokogiri
end
```

**Comment by ronaldtse** [comment]
Actually, custom methods is not working for XML:
* https://github.com/lutaml/lutaml-model/pull/45

We need to fix this.

**Comment by ronaldtse** [comment]
The issue is fixed and released in 0.3.5.
